drivers/hmis/app/models/hmis/staff_assignment.rb

```diff
@@ -11,4 +11,9 @@ class Hmis::StaffAssignment < Hmis::HmisBase
   belongs_to :user, class_name: 'Hmis::User'
   belongs_to :household, class_name: 'Hmis::Hud::Household', primary_key: [:data_source_id, :HouseholdID], foreign_key: [:data_source_id, :household_id], inverse_of: :staff_assignments
   belongs_to :staff_assignment_relationship, foreign_key: :hmis_staff_assignment_relationship_id, class_name: 'Hmis::StaffAssignmentRelationship'
+
+  scope :viewable_by, ->(user) do
+    # Special case this, rather than using EnrollmentRelated concern, because we have to join through Household
+    joins(household: :enrollments).merge(Hmis::Hud::Enrollment.viewable_by(user))
```

Comment:
Theron made some perf improvements to the Household.viewable_by scope [here](https://github.com/greenriver/hmis-warehouse/pull/3602/files#diff-e9d936441ab6d0722484ec8a11511424d896354709c10ad718fd658cf317a955) by using a correlated subquery. 
Since this loads on the home page for some users, I think we should look at perf. Is this faster?:
```
joins(:household).merge(Hmis::Hud::Household.viewable_by(user)
```

Reply:
They are comparable in speed, but the original version had a bug which your update fixes. I also think your update is clearer. Thanks!

They are not too fast, 6 seconds on average for me locally. I'd like to check it out on QA.

Bug was: When joining to enrollments, the dashboard table would show n rows per household where n is the number of enrollments in the household (but all rows would display the HoH's name and link to the HoH's enrollment).

<details>
<summary>timing details</summary>

```
joins(:household).merge(Hmis::Hud::Household.viewable_by(user))
```
- Completed 200 OK in 6143ms (Views: 0.7ms | ActiveRecord: 5920.9ms | Allocations: 259556)
- Completed 200 OK in 6312ms (Views: 0.6ms | ActiveRecord: 6124.2ms | Allocations: 213006)
- Completed 200 OK in 5948ms (Views: 1.0ms | ActiveRecord: 5794.0ms | Allocations: 94312)

```
joins(household: :enrollments).merge(Hmis::Hud::Enrollment.viewable_by(user))
```
- Completed 200 OK in 5938ms (Views: 2.3ms | ActiveRecord: 5720.6ms | Allocations: 247556)
- Completed 200 OK in 5983ms (Views: 8.4ms | ActiveRecord: 5763.1ms | Allocations: 128526)
- Completed 200 OK in 6422ms (Views: 3.9ms | ActiveRecord: 6231.3ms | Allocations: 128522)
</details>

Reply:
okay good catch. sounds good re: testing perf on qa.